src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,165 @@
 package core.basesyntax;
 
-public class ArrayList<T> implements List<T> {
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
```

Comment:
Don't use Objects, Arrays, or any other util class. Instead of using Objects.equals(), you can compare the elements for null and equality separately.